### PR TITLE
Feature add referrals

### PR DESCRIPTION
**Pull request by samgreen**
We also have a basic implementation of Universal Links working with `https://fallball.io`

This [Document](https://developer.apple.com/documentation/technotes/tn3155-debugging-universal-links) was very helpful.

View the [apple-app-site-association](https://fallball.io/.well-known/apple-app-site-association)

- [x] Integrate Remote Config
- [ ] Figure out backend rewarding

### CODE REVIEW

File: endlessfaller.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1084,7 +1134,7 @@
 				"LD_RUNPATH_SEARCH_PATHS[sdk=macosx*]" = "@executable_path/../Frameworks";
 				MACOSX_DEPLOYMENT_TARGET = 13.3;
 				MARKETING_VERSION = 1.7.7;
-				PRODUCT_BUNDLE_IDENTIFIER = io.endlessfall;
+				PRODUCT_BUNDLE_IDENTIFIER = io.endlessfall.sam;
```
Comment (samgreen) [inline endlessfaller.xcodeproj/project.pbxproj]: These need to be reverted

**Review by we-z — COMMENTED** [review]
add new generated referral link to line 117 in PlayersPlaqueView

**Review by we-z — COMMENTED** [review]
hell ya

**Comment by samgreen** [comment]
Test these links:

1. [https://fallball.io/link?ref=SAMMY](https://fallball.io/link?ref=SAMMY)
2. [io-fallball://install?ref=SAMMY](io-fallball://install?ref=SAMMY)

**Comment by samgreen** [comment]
Smart banners

image

File: endlessfaller/Configs/UserData.swift [hunk 2]
```diff
@@ -20,10 +21,12 @@ class UserPersistedData: ObservableObject {
     @CloudStorage("boinIntervalCounter") var boinIntervalCounter: Int = 0
     @CloudStorage("infiniteBoinsUnlocked") var infiniteBoinsUnlocked: Bool = false
     @CloudStorage("strategyModeEnabled") var strategyModeEnabled: Bool = false
+    @CloudStorage("referralURL") var referralURL: String = ""
```
Comment (samgreen) [inline endlessfaller/Configs/UserData.swift:24]: Remove this

File: endlessfaller/endlessfallerApp.swift [hunk 3]
```diff
@@ -6,15 +6,24 @@
 //
 
 import SwiftUI
+import FirebaseRemoteConfig
```
Comment (samgreen) [inline endlessfaller/endlessfallerApp.swift:9]: Remove this

**Review by we-z — COMMENTED** [review]
boins coming soon

**Pull request merged**